File: Snowman.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Wed Apr 20 18:22:00 2022

@author: bushra

------------------------------------------------------------------------------
   Name:     BI_IS221_PP_3_10
   Author:   Bushra Ibrahim
   Date:     Apr 20, 2022
   Language: Python
   Purpose:  The purpose of this program is to draw a Snowman based on
             the edits mentioned in PP 3.10.
------------------------------------------------------------------------------
   ChangeLog:
   
   Who:      Bushra Ibrahim           
   Date:     Apr 20, 2022
   Desc.:    This is the original version of the code.
------------------------------------------------------------------------------

"""


from graphics import *

win = GraphWin('Snowman',500,500)
win.setCoords(-100,-100,100,100)
win.setBackground('light blue')

rctGround = Rectangle(Point(-100, -50), Point(100,-100))
rctGround.setFill('dark blue')
rctGround.setOutline('dark blue')
rctGround.draw(win)

crcSun = Circle(Point(75,60), 15)
crcSun.setFill('gold')
crcSun.setOutline('gold')
crcSun.draw(win)

crcSnow1 = Circle(Point(30,-25), 35)
crcSnow1.setFill('white')
crcSnow1.setOutline('white')
crcSnow1.draw(win)

crcSnow2 = Circle(Point(30,15), 25)
crcSnow2.setFill('white')
crcSnow2.setOutline('white')
crcSnow2.draw(win)

crcButton1 = Circle(Point(30,25), 2.5)
crcButton1.setFill('red')
crcButton1.setOutline('red')
crcButton1.draw(win)
crcButton1 = Circle(Point(30,15), 2.5)
crcButton1.setFill('red')
crcButton1.setOutline('red')
crcButton1.draw(win)
crcButton1 = Circle(Point(30,5), 2.5)
crcButton1.setFill('red')
crcButton1.setOutline('red')
crcButton1.draw(win)


crcSnow3 = Circle(Point(30, 50), 15)
crcSnow3.setFill('white')
crcSnow3.setOutline('white')
crcSnow3.draw(win)

crcEye1 = Circle(Point(25,55), 3)
crcEye1.setFill('black')
crcEye1.setOutline('black')
crcEye1.draw(win)
crcEye1 = Circle(Point(35,55), 3)
crcEye1.setFill('black')
crcEye1.setOutline('black')
crcEye1.draw(win)

lnSmile = Line(Point(25,45),Point(35,45))
lnSmile.setFill('black')
lnSmile.draw(win)

lnArm1 = Line(Point(15,15),Point(-10,25))
lnArm1.setFill('black')
lnArm1.draw(win)

lnArm2 = Line(Point(45,15),Point(70,25))
lnArm2.setFill('black')
lnArm2.draw(win)

txtName = Text(Point(-45,70),'Bushra Ibrahim')
txtName.setFill('white')
txtName.draw(win)

rctHat1 = Rectangle(Point(10, 62), Point(50,60))
rctHat1.setFill('black')
rctHat1.setOutline('black')
rctHat1.draw(win)

rctHat2 = Rectangle(Point(20, 80), Point(40,60))
rctHat2.setFill('black')
rctHat2.setOutline('black')
rctHat2.draw(win)


win.getMouse()
win.close()

File: Snowman.java
/*
------------------------------------------------------------------------------
   Name:     Snowman
   Author:   Bushra Ibrahim
   Date:     Feb 16, 2022
   Language: Java
   Purpose:  The purpose of this program is to make a graphic window displaying
             a snowman, but make the edits mentioned in PP 3.10.
------------------------------------------------------------------------------
   ChangeLog:
   Who:      Bushra Ibrahim           Date:     Feb 16, 2022
   Desc.:    This is a modified version of the original Snowman.java code.
------------------------------------------------------------------------------
*/

package snowman;

import javafx.application.Application;
import javafx.stage.Stage;
import javafx.scene.Group;
import javafx.scene.Scene;
import javafx.scene.paint.Color;
import javafx.scene.shape.*;
import javafx.scene.text.Text;

public class Snowman extends Application
        
{
    //--------------------------------------------------------
    //  Presents a snowman scene.
    //--------------------------------------------------------
    @Override
    public void start(Stage primaryStage)
//************************************************************
 {
    Ellipse base = new Ellipse(80, 210, 80, 60);
    base.setFill(Color.WHITE);
    
    Ellipse middle = new Ellipse(80, 130, 50, 40);
    middle.setFill(Color.WHITE);
    
    Circle head = new Circle(80, 70, 30);
    head.setFill(Color.WHITE);
    
    Circle rightEye = new Circle(70, 60, 5);
    
    Circle leftEye = new Circle(90, 60, 5);
    
    Line mouth = new Line(70, 80, 90, 80);
    
    Circle topButton = new Circle(80, 110, 6);
    topButton.setFill(Color.RED);
    
    Circle middleButton = new Circle(80, 125, 6);
    middleButton.setFill(Color.RED);
    
    Circle bottomButton = new Circle(80, 140, 6);
    bottomButton.setFill(Color.RED);
    
    Line leftArm = new Line(110, 130, 160, 130);
    leftArm.setStrokeWidth(3);
    
    Line rightArm = new Line(50, 130, 0, 100);
    rightArm.setStrokeWidth(3);
    
    Rectangle stovepipe = new Rectangle(60, 0, 40, 50);
    
    Rectangle brim = new Rectangle(50, 45, 60, 5);
    
    Group hat = new Group(stovepipe, brim);
    
    hat.setTranslateX(10);
    
    hat.setRotate(15);
    
    Group snowman = new Group(base, middle, head, leftEye, rightEye, mouth, 
                              topButton, middleButton, bottomButton, leftArm, 
                              rightArm, hat); //mid button added
    
    snowman.setTranslateX(210); //40 pixel move
    snowman.setTranslateY(50);
    
    Circle sun = new Circle(400, 50, 30); //moved location of sun
    sun.setFill(Color.GOLD);
    
    Text name = new Text(50,50,"Bushra Ibrahim"); //added my name to top left
    name.setFill(Color.SNOW);
    
    Rectangle ground = new Rectangle(0, 250, 500, 100);
    ground.setFill(Color.STEELBLUE);
    
    Group root = new Group(ground, sun, snowman, name);
    Scene scene = new Scene(root, 500, 350, Color.LIGHTBLUE);
    
    primaryStage.setTitle("Snowman");
    primaryStage.setScene(scene);
    primaryStage.show(); 
}
    public static void main(String[] args)
    {
        launch(args);
    }
}
